bull/google-books-queue/consumer.js

```diff
@@ -88,12 +88,16 @@ GoogleBooksQueue.process((job, done) => {
               level: "error",
               message: `IA Failure GB ${body}`,
             });
+            if (job.data.isEmailNotification === "true") {
+              EmailProducer(job.data.userName, title, trueURI, false);
```

Comment:
What if it failed at `if (!body) {`? Keep it outside the else block and call the function once in `if (error || response.statusCode != 200) {` block.